src/readBuffer.js

```diff
@@ -23,10 +30,18 @@ export default function readBuffer(pipe, length, callback) {
         if (overflow) {
           pipe.unshift(overflow);
         }
+        terminated = true;
         callback(null, Buffer.concat(buffers, length));
       }
     };
     pipe.on('data', onChunk);
+    pipe.on('end', function() {
+      if (terminated) {
+        return;
+      }
+      terminated = true;
+      callback(EOF);
```

Comment:
Is EOF an error condition?

Reply:
Yes, you hit it if you try to read `length` bytes from the stream but the stream ends before you get that many.